src/string.c

```diff
@@ -206,3 +206,43 @@ void printf(const char *s, ...) {
     }
     va_end(args);
 }
+
+void to_upper(char *dest, const char *src) {
+    int i;
+    for (i = 0; i < strlen(src); i++) {
+        if (src[i] >= 97 && src[i] <= 122) {
+            dest[i] = src[i] - 32;
+        } else {
+            dest[i] = src[i];
+        }
+    }
+    dest[strlen(src)] = '\0';
```

Comment:
Yes. `dest` is an empty char buffer, so we need to null terminate it. It's also the exact same size as `src` so we know where to null terminate it.